src/filters/renderer/VideoRenderers/DX9RenderingEngine.cpp

```diff
@@ -390,20 +390,28 @@ HRESULT CDX9RenderingEngine::RenderVideoDrawPath(IDirect3DSurface9* pRenderTarge
             start = stop;    // reset after 10 min (ps float has its limits in both range and accuracy)
         }
 
+        LONGLONG lCurrentPosition = 0;
+        LONGLONG lDuration = 0;
+        CComPtr<IMediaStatusPresenter> pInfos;
+        if (SUCCEEDED(GetMediaStatusPresenter(&pInfos))) {
+            pInfos->GetMediaTimes(&lCurrentPosition, &lDuration);
+        }
+        const float clTickToSeconds = 1.0f / 10000000.0f;
+
 #if 1
         D3DSURFACE_DESC desc;
         m_pVideoTexture[m_nCurSurface]->GetLevelDesc(0, &desc);
 
         float fConstData[][4] = {
-            {(float)desc.Width, (float)desc.Height, (float)(counter++), (float)diff / CLOCKS_PER_SEC},
-            {1.0f / desc.Width, 1.0f / desc.Height, 0, 0},
+            { (float)desc.Width, (float)desc.Height, (float)(counter++), (float)diff / CLOCKS_PER_SEC },

```

Comment:
If I understand correctly you are trying to give the shaders access to the
current frame time. I'm actually not sure that what you do is correct. It
can sort of work but it seems hacky to rely on the info gathered from
CMainFrame. There must be cleaner, safer, more accurate way to do that
directly from the renderer.